kube-client/src/client/client_ext.rs

```diff
@@ -0,0 +1,258 @@
+/// // List at Cluster level:
+/// for pod in client.list::<Pod>(&lp, &Cluster).await? {
+///     println!("Found pod {} in {}", pod.name_any(), pod.namespace().unwrap());
+/// }
+/// // Namespaced Get:
+/// let svc = client.get::<Service>("kubernetes", &Namespace::from("default")).await?;
+/// assert_eq!(svc.name_unchecked(), "kubernetes");
+/// # Ok(())
+/// # }
+/// ```
+impl Client {
+    /// Get a single instance of a `Resource` implementing type `K` at the specified scope.
+    ///
+    /// ```no_run
+    /// # use k8s_openapi::api::rbac::v1::ClusterRole;
+    /// # use k8s_openapi::api::core::v1::Service;
+    /// # use kube::client::scope::{Cluster, Namespace};
+    /// # use kube::{ResourceExt, api::GetParams};
+    /// # async fn wrapper() -> Result<(), Box<dyn std::error::Error>> {
+    /// # let client: kube::Client = todo!();
+    /// let cr = client.get::<ClusterRole>("cluster-admin", &Cluster).await?;
+    /// assert_eq!(cr.name_unchecked(), "cluster-admin");
+    /// let svc = client.get::<Service>("kubernetes", &Namespace::from("default")).await?;
+    /// assert_eq!(svc.name_unchecked(), "kubernetes");
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn get<K>(&self, name: &str, scope: &impl ObjectUrl<K>) -> Result<K>
+    where
+        K: Resource + Serialize + DeserializeOwned + Clone + Debug,
+        <K as Resource>::DynamicType: Default,
+    {
+        let mut req = Request::new(scope.url_path())
+            .get(name, &GetParams::default())
+            .map_err(Error::BuildRequest)?;
+        req.extensions_mut().insert("get");
+        self.request::<K>(req).await
+    }
+
+    /// List instances of a `Resource` implementing type `K` at the specified scope.
+    ///
+    /// ```no_run
+    /// # use k8s_openapi::api::core::v1::Pod;
+    /// # use k8s_openapi::api::core::v1::Service;
+    /// # use kube::client::scope::{Cluster, Namespace};
+    /// # use kube::{ResourceExt, api::ListParams};
+    /// # async fn wrapper() -> Result<(), Box<dyn std::error::Error>> {
+    /// # let client: kube::Client = todo!();
+    /// let lp = ListParams::default();
+    /// for pod in client.list::<Pod>(&lp, &Cluster).await? {
+    ///     println!("Found pod {} in {}", pod.name_any(), pod.namespace().unwrap());
+    /// }
+    /// for svc in client.list::<Service>(&lp, &Namespace::from("default")).await? {
+    ///     println!("Found service {}", svc.name_any());
+    /// }
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn list<K>(&self, lp: &ListParams, scope: &impl CollectionUrl<K>) -> Result<ObjectList<K>>
+    where
+        K: Resource + Serialize + DeserializeOwned + Clone + Debug,
+        <K as Resource>::DynamicType: Default,
+    {
+        let mut req = Request::new(scope.url_path())
+            .list(lp)
+            .map_err(Error::BuildRequest)?;
+        req.extensions_mut().insert("list");
+        self.request::<ObjectList<K>>(req).await
+    }
+}
+
+#[cfg(test)]
+#[cfg(feature = "client")]
```

Comment:
good catch. removed this inner test gate entirely, so that it is always has the same inclusion rules as the module. https://github.com/kube-rs/kube/pull/1375/commits/44a740bce2a857bc8c3be7bfec8d25021ceff84a